### ISSUE TITLE
Is it possible to modify the parameters in GPT-2?

### ISSUE THREAD

**Issue by weiguowilliam** [huggingface/transformers#1403]
## ❓ Questions & Help

I wonder whether it's possible to modify the parameters in GPT-2? Since we can not train GPT-2, modifying the parameters and observing the changes in results will be helpful. Thank you in advance!

**Comment by LysandreJik** [comment]
Hi! GPT-2, like all models in this library, directly inherit from pytorch's `nn.Module`, so you're free to finetune them or modify their parameters as you wish.

**Comment by weiguowilliam** [comment]
> Hi! GPT-2, like all models in this library, directly inherit from pytorch's `nn.Module`, so you're free to finetune them or modify their parameters as you wish.

Hi Lysandre,

Thank you for your reply. I tried with model.state_dict() to modify the weight matrix. But it shows no change in output. Can you help me check that? I'm new to Pytorch. Thank you in advance!

"""
#initiliazation
tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
model = GPT2LMHeadModel.from_pretrained('gpt2')
keys = model.state_dict()

#modify parameter
i = 0
for k,v in keys.items():
    if i == 5:
        print(keys[k])
        a  = v
        keys[k] = torch.rand(a.shape)
        print(keys[k])    
    i+=1

#prediction
text = tokenizer.encode("The teacher is a")
input, past = torch.tensor([text]), None
logits, past = model(input, past = past)
logits = logits[0,-1]
probabilities = torch.nn.functional.softmax(logits)
best_logits, best_indices = logits.topk(10)
best_words = [tokenizer.decode([idx.item()]) for idx in best_indices]
text.append(best_indices[0].item())
best_probabilities = probabilities[best_indices].tolist()
print(best_words)
"""

I use random matrix to replace 'h.0.attn.c_attn.weight'. And I used the modified model to make prediction. However, the output of model doesn't change. I tried to simultaneously change 10,000 matrix to random number. But it still doesn't work. Do I misunderstand how pytorch works?

**Comment by weiguowilliam** [comment]
> Hi! GPT-2, like all models in this library, directly inherit from pytorch's `nn.Module`, so you're free to finetune them or modify their parameters as you wish.

Thank you for your help!
